internal/appconfig/serde_test.go

```diff
@@ -602,3 +602,33 @@ func TestIsSameTOMLAppConfigReferenceFormat(t *testing.T) {
 	actual.configFilePath = ""
 	require.Equal(t, cfg, actual)
 }
+
+// bin/flyctl config show --local -c internal/appconfig/testdata/full-reference.toml > internal/appconfig/testdata/full-reference.json
+func TestIsSameJSONAppConfigReferenceFormat(t *testing.T) {
+	const TOMLpath = "./testdata/full-reference.toml"
+	TOMLcfg, err := LoadConfig(TOMLpath)
+	require.NoError(t, err)
+
+	const JSONpath = "./testdata/full-reference.json"
```

Comment:
Sam, feel free to ignore this nit, here it goes:

As far as I see, we are not testing `WriteTo()` and (I know I asked for it) by adding the json and yaml versions of full-reference, we have another set of files to maintain which is good and even after below suggestion I don't think we have to change that. 

All that to say we should be serializing the unmarshaled toml into toml, json and yaml; and then loading from this serialized files to compare the configs.

Reply:
Not a nit.

I removed those new files, and changed the existing tests to start from the full-reference.toml file and write the config out and read it back in for both json and yaml formats.... and found a bug.  I've been focusing mostly on `fly config show` and visual inspection, but `WriteTo` will add an extra comment at the top - and JSON does not allow comments.   So by adding a test, I found and fixed a but.

Thanks!